File: elapsedTimeLogger.go
package main

import (
	"fmt"
	"time"
)

func elapsedTimeLogger(namespace string) func(string) {
	start := time.Now().UnixNano()
	return func(msg string) {
		now := time.Now().UnixNano()
		fmt.Println(fmt.Sprintf("%s: %s took %d millis", namespace, msg, (now-start)/1000000))
		start = now
	}
}

File: elapsed-time-logger.rs
pub fn elapsed_time_logger(namespace: &str) -> impl FnMut(&str) {
    let namespace = namespace.to_owned();
    let mut start = std::time::Instant::now();
    move |message| {
        let duration = start.elapsed().as_millis();
        println!("{namespace}: {message} took {duration} millis",);
        start = std::time::Instant::now();
    }
}
